extensions/amp-mowplayer/0.1/amp-mowplayer.js

```diff
@@ -228,6 +230,10 @@ class AmpMowplayer extends AMP.BaseElement {
 
     const {element} = this;
 
+    if (eventType === 'set_aspect_ratio') {
```

Comment:
This equates to a `height` change which is not always allowed. `attemptChangeHeight` is the API to use here.

@gopanisandip could could you explain what this PR is trying to do?

Reply:
It looks like this is doing something very similar to `layout=responsive`? `padding-top` hack and all :)

Reply:
@alanorozco @aghassemi  Let me explain what we need is :  We have player with playlist so in that case aspect ration of player is not always 16:9 because we have feature in mowplayer.com so that user can choose position of playlist like right side of player and bottom of player. so in that case we have 2 different aspect ratio.  Also if user choose right position of playlist than also we have it in bottom in mobile device.

so when ever player load or screen rotate at that time we event will be fired from iframe with new aspect ratio  and in this way we can make it responsive.  This event will calculate padding-Top based on internal content of player.

This is not limited to playlist only we have another version of player which have different aspect ration based on screen resolution aspect ration will change.

Please check this screenshot links :

http://prntscr.com/ltbzev
http://prntscr.com/ltbzpq
http://prntscr.com/ltbzu3
http://prntscr.com/ltbzxw
http://prntscr.com/ltc030

Could help me to figure out this? Problem here is we don't have fixed aspect ratio.

Reply:
Got it. See @aghassemi's comment above. You need to calculate the fixed height based on the component's width (basically `height = width * scale`, where scale is the value received as 'padding-top' divided by `100`).

Then you can `attemptChangeHeight(height)`. The runtime may reject the request to change height, and that would be working as intended (It's one of AMP's core promises that content should not shift vertically under certain conditions).

Reply:
Got it.   Thanks  will change it!